### DISCUSSION TITLE
Server snippet not reflecting in nginx.conf file

### DISCUSSION THREAD

**Discussion by Asiya-Yunusa** [nginx/kubernetes-ingress#6923] [Q&A]
Hello, I am trying to set two variable in the my nginx.conf server block. I added the server-snippets annotation to the nginx deployment file but it is not reflecting. When i add the snippet to a particular ingress resource it reflects but the variable value does not populate.

This is the annotation in the nginx deployment which doesn't change the config file
```
    annotations:
        nginx.org/server-snippets: |
          set $http_x_app_name $http_x_app_name;
          set $http_x_country_code $http_x_country_code;
```
The annotation above  is also what I add to the ingress resource that populates the config file as below, all other parameters set in the main nginx.conf file are populated

![image](https://github.com/user-attachments/assets/af869b2b-bab5-4438-b69e-cc8c634c37b4)

**Comment by pdabelf5** [comment]
@Asiya-Yunusa The `http_` variable prefix is reserved for [NGINX header variables](https://nginx.org/en/docs/http/ngx_http_core_module.html#var_http_) and cannot be set directly.  Apologies for the delay responding to your question.
